std/experimental/color/rgb.d

```diff
@@ -0,0 +1,852 @@
+module std.experimental.color.rgb;
+
+import std.experimental.color;
+import std.experimental.color.xyz;
+
+import std.traits: isInstanceOf, isNumeric, isIntegral, isFloatingPoint, isSigned, isSomeChar, Unqual;
+import std.typetuple: TypeTuple;
+import std.typecons: tuple;
+
+@safe: pure: nothrow: @nogc:
+
+
+enum isRGB(T) = isInstanceOf!(RGB, T);
+
+enum RGBColorSpace
+{
+    sRGB,
+    sRGB_Gamma2_2,
+/+ TODO: these all need testing
+    AdobeRGB,
+    AppleRGB,
+    BestRGB,
+    BetaRGB,
+    BruceRGB,
+    CIE_RGB,
+    ColorMatchRGB,
+    DonRGB4,
+//    ECI_RGB_v2,
+    EktaSpacePS5,
+    NTSC_RGB,
+    PAL_SECAM_RGB,
+    ProPhotoRGB,
+    SMPTE_C_RGB,
+    WideGamutRGB
++/
+}
+
+
+struct RGB(string components_, ComponentType_, bool linear_ = false, RGBColorSpace colorSpace_ = RGBColorSpace.sRGB) if(isValidComponentType!ComponentType_)
```

Comment:
I might note, a sort-of exception exists for sRGB, which is often encoded with gamma 2.2 rather than it's proper (ridiculous!) ramp, usually for reasons of efficiency. I've expressed this distinction by a separate colourspace sRGB2_2.